### DISCUSSION TITLE
[ArzeshiYaab / Qassem Check for Twitter] Chrome extension for twitter to identify potential members who are from the I.R. cyber army.

### DISCUSSION THREAD

**Discussion by HosseinYousefi** [WeAreMahsaAmini/Core#416] [Ideas]
# ArzeshiYaab / Qassem Check for Twitter

## Motivation

Twitter is a great space for open discussion, however it's filled with disinformation and propaganda from "Iranian Cyber Army" spreading lies.

## Solution

The idea is to create a simple web app to check the suspiciousness of a twitter handle. Alongside with mechanisms for reporting. At the beginning a team of trusted people could check these reports. Later on, we could introduce automatic recognition and checking (e.g. using NLP.)

A Chrome extension could be used to directly overlay these reports on twitter.com. 

A sample visualisation of such extension:

<img width="408" alt="Screen Shot 2022-10-06 at 13 29 16" src="https://user-images.githubusercontent.com/9751815/194301576-fc8776e9-9dbf-4952-980b-62c0b48e76b1.png">

## Team

I'm ready to be working on this in my free time and lead the project.

**Comment by ZanZendegiAzadi401** [comment]
**WIP**

## Sus user detection and reporting

#### Citizen Reports
Each Twitter user can report a potential Ghasem Check Positive account by replying to the following under a suspicious tweet: 

- **@GhasemCheck Arzeshi** _(to report Arzeshi accounts)_ 
- **@GhasemCheck Cyberi** _(To report potential Cyberi accounts)_ 

The reported account will then be added to Ghasem Check's databases, pending verification

#### Auto reporting

-
-


## Verification

Each User handle has a sus rating 

- Sus rating < 25 -- Safe
- Sus rating > 26 -- Sus
- Sus rating > 70 -- Unsafe

- (how do you identify if they are Arzeshi or Cyberi? needs another layer) 

#### Auto verification
##### Obvious metrics to increase sus rating for each handle: 

- Twitter join date especially post sept. 2022 
- Follower/tweet ratio
- Followers and following -- are they sus too? 
- Keywords in tweets 
- key Hashtags 
- tweets per day / most activity at a specific time of day 
- photo of Ghasem or Khamenei in the banner, profile, or frequently posted 
- (feel free to add more) 
- Number of citizen reports 
- 
##### Not-so-obvious metrics:
- 
-


#### Human Verification
- a trusted human can check for said metrics and determine the score 
- a trusted human can also use common sense lol

## Post verification 

Once verified as unsafe, and categorised under Arzeshi or Cyberi:

- Handle will be introduced on @GhasemCheck Twitter page 
- Handle will be added to the blacklist database. 
- Ghasem Check rating should display above each tweet and on the user's profile page 

## Blacklist
(Technically possible?) There should be 2 blacklist to automatically block all identified 1- Arzeshi, 2-Cyberi, accounts. Such list should be automatically added to a user's blocked users list once they click a link (auth needed I guess? have no idea how this would work)

**Comment by grimpenmire** [comment]
Just a quick note: I think a telegram link (usually an "anonymous" chat; which is not necessarily anonymous) in bio is also a very good indicator of the so called "cyberi" accounts.

**Comment by grimpenmire** [comment]
Another idea. It would be nice that each user could have an account and keep their own list of suspects. These lists can then be aggregated, taking into account the trustworthiness of the user themselves, to create global lists.

**Comment by grimpenmire** [comment]
Gathering lists of likers and retweeters (of known criminals!) can also help with scoring. Overall, this has to potential to become a cool big data project. Unfortunately, that also means it likely needs a good amount of resources.

**Comment by ghost** [comment]
https://github.com/WeAreMahsaAmini/Core/discussions/256
What do you think?
I have the database and algorithem but not the knowleg nessesary for development. If intrested contact me please.

**Reply by HosseinYousefi** [reply]
> https://github.com/WeAreMahsaAmini/Core/discussions/256
> 
> What do you think?
> 
> I have the database and algorithem but not the knowleg nessesary for development. If intrested contact me please.

I think autoblocking complicates our efforts initially but is definitely possible to do. Other than that, I have the expertise required to develop this. I'm currently seeing if this is actually useful in these times for people or not.

**Reply by ghost** [reply]
Let me know how we can have chat or talk, however works for you.

Im pretty sure one immidiate application is very helpful for users, and that is refining their folowers in such a situation.

**Comment by ma21ar** [comment]
I was planning to implement a similar project and I will be happy to help you finish this one.
I'm a software developer and can help you with the coding part.

**Comment by TheTraxex** [comment]
I found a potentially useful link which explains the logic to identify the fake accounts:
[https://farokhnotes.medium.com/twitters-war-against-fake-accounts-here-s-how-to-win-it-as-google-did-2db59ce85465](url)

Basically the author introduced two main criteria to determine if a twitter account is fake or not. One is "the action of each account" and another is "the interaction of that account with other accounts" and for both of them he suggested ways to calculate them.

**Comment by pachol1401** [comment]
Please have a look on the comment here:
https://github.com/WeAreMahsaAmini/Core/discussions/256#discussioncomment-3833724
